src/pci/src/configuration.rs

```diff
@@ -924,9 +924,15 @@ impl PciConfiguration {
                     region_type,
                 });
             } else if (reg_idx > BAR0_REG)
-                && ((self.registers[reg_idx - 1] & self.writable_bits[reg_idx - 1])
-                    != (self.bars[bar_idx - 1].addr & self.writable_bits[reg_idx - 1])
-                    || (value & mask) != (self.bars[bar_idx].addr & mask))
+                && (
+                    // The lower BAR (of this 64bit BAR) has been reprogrammed to a different value
+                    // than it used to be
+                    (self.registers[reg_idx - 1] & self.writable_bits[reg_idx - 1])
+                    != (self.bars[bar_idx - 1].addr & self.writable_bits[reg_idx - 1]) ||
+                    // Or the lower BAR hasn't been changed but the upper one is being reprogrammed
+                    // now to a different value
+                    (value & mask) != (self.bars[bar_idx].addr & mask)
+                )
```

Comment:
Shouldn't this be easier to check by doing `old_base != new_base`? Actually same can be dones for 32bit BARs as well.

Reply:
the `old_base != new_base` would cover both. Bellow we calculate `old/new_base` as `u64`s, so both BARs are in use, so it will catch the case where any BAR changed.